src/engine/blocks.js

```diff
@@ -288,6 +290,24 @@ class Blocks {
             break;
         case 'checkbox':
             block.isMonitored = args.value;
+            if (optRuntime && wasMonitored && !block.isMonitored) {
+                optRuntime.removeMonitors([{id: block.id}]);
+            } else if (optRuntime && !wasMonitored && block.isMonitored) {
+                optRuntime.addMonitors(
+                    // Ensure that value is not undefined, since React requires it
+                    [{
+                        // @todo(dd) this will collide if multiple sprites use same block
+                        id: block.id,
+                        category: 'data',
+                        // @todo(dd) how to handle translation here?
+                        label: block.opcode,
+                        value: '',
+                        x: 0,
+                        // @todo(dd) place below the last monitor instead
+                        y: 0
+                    }]
+                );
+            }
```

Comment:
Firing the add/remove events here makes the GUI need to do more management than necessary. Let's try to only emit the current state of the monitor list each tick.